File: ngDailyTrainer/src/app/services/comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { catchError, throwError } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Trade } from '../models/trade';
import { AuthService } from './auth.service';

@Injectable({
  providedIn: 'root'
})
export class CommentService {

  private url = environment.baseUrl + 'api/comments'


  constructor(private http: HttpClient, private auth: AuthService) { }

  index(trades:Trade[]) {
    return this.http.get<Comment[]>(this.url, this.auth.getHttpOptions()).pipe(
      catchError((err:any) => {
        console.log(err);
        return throwError(() => new Error('KABOOM - comment list cannot be retrieved.'));
      }));

}
}

File: day-trader-monolith/DayTrainer/src/main/java/com/skilldistillery/daytrainer/comment/CommentService.java
package com.skilldistillery.daytrainer.comment;

import java.util.List;
import com.skilldistillery.daytrainer.entities.Comment;
import com.skilldistillery.daytrainer.entities.Trade;

public interface CommentService {

	public List<Comment> getComment(List<Trade> trades);

}
